client/ayon_core/tools/push_to_project/main.py

```diff
@@ -4,14 +4,20 @@
 from ayon_core.tools.push_to_project.ui import PushToContextSelectWindow
 
 
-def main_show(project_name, version_id):
-    app = get_ayon_qt_app()
-
-    window = PushToContextSelectWindow()
+def main_show(project_name, version_id, library_filter, context_only):
+    window = PushToContextSelectWindow(
+        library_filter=library_filter, context_only=context_only
+    )
     window.show()
     window.set_source(project_name, version_id)
 
-    app.exec_()
+    if __name__ == "__main__":
+        app = get_ayon_qt_app()
```

Comment:
Creation of window may require already existing `QApplication`, otherwise will created new (which is something we don't want to).

Reply:
If you need functionality, which looks like it is not in fact "main" please create new function.

Reply:
Cool, done.